packages/quick_actions/quick_actions_ios/ios/Classes/QuickActionsPlugin.swift

```diff
@@ -4,48 +4,44 @@
 
 import Flutter
 
-public final class QuickActionsPlugin: NSObject, FlutterPlugin {
+public final class QuickActionsPlugin: NSObject, FlutterPlugin, IOSQuickActionsApi {
 
   public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(
-      name: "plugins.flutter.io/quick_actions_ios",
-      binaryMessenger: registrar.messenger())
-    let instance = QuickActionsPlugin(channel: channel)
-    registrar.addMethodCallDelegate(instance, channel: channel)
+    let messenger = registrar.messenger()
+    let instance = QuickActionsPlugin(messenger: messenger)
+    IOSQuickActionsApiSetup.setUp(binaryMessenger: messenger, api: instance)
     registrar.addApplicationDelegate(instance)
   }
 
-  private let channel: MethodChannel
   private let shortcutItemProvider: ShortcutItemProviding
-  private let shortcutItemParser: ShortcutItemParser
+  private let shortcutItemParser: ShortcutItemParser = DefaultShortcutItemParser()
+  private let flutterApi: IOSQuickActionsFlutterApiProtocol
   /// The type of the shortcut item selected when launching the app.
   private var launchingShortcutType: String? = nil
 
+  // This init is meant for unit testing only.
```

Comment:
A comment isn't good enough to prevent it from being misused in non-unit test scenario. I suggest just keep this init function, and remove the convenient init below, and then inject `IOSQuickActionsFlutterApi(binaryMessenger: messenger)` from `static func register(with:)`.